src/handlers/commands/ToggleDeployEnablement.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright © 2018 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    failure,
+    HandleCommand,
+    MappedParameter,
+    MappedParameters,
+    Success,
+} from "@atomist/automation-client";
+import { Parameters } from "@atomist/automation-client/decorators";
+import { HandlerContext } from "@atomist/automation-client/Handlers";
+import { commandHandlerFrom } from "@atomist/automation-client/onCommand";
+import { addressEvent } from "@atomist/automation-client/spi/message/MessageClient";
+import {
+    DeployEnablement,
+    DeployEnablementRootType,
+} from "../../ingesters/deployEnablement";
+import { success } from "../../util/slack/messages";
+
+@Parameters()
+export class ToggleDeployEnablementParameters {
+
+    @MappedParameter(MappedParameters.GitHubOwner)
+    public owner: string;
+
+    @MappedParameter(MappedParameters.GitHubRepository)
+    public repo: string;
+
+    @MappedParameter(MappedParameters.GitHubRepositoryProvider)
+    public providerId: string;
+
+}
+
+function toggleEnablement(enable: boolean) {
```

Comment:
Can we call this setEnablement or saveEnablement? It is not a toggle
Which makes me happy, toggles are evil

Reply:
For sure. I’ll rename as you like.